iml-gui/crate/src/components/stratagem/mod.rs

```diff
@@ -1,68 +1,455 @@
+            Msg::CheckStratagem => {
+                if !can_enable(&model.fs, cache) {
+                    orders.send_msg(Msg::DisableStratagem);
+                }
+            }
+            Msg::DisableStratagem => {
+                model.state = State::Disabled;
+            }
+            Msg::EnableStratagem => {}
+            Msg::SendCommand(cmd) => {
+                config.disabled = true;
+
+                match cmd {
+                    Command::Update => {
+                        if let Some(x) = config.get_stratagem_update_config(model.fs.id) {
+                            config.target_config = TargetConfig {
+                                interval: x.interval,
+                                report_duration: x.report_duration,
+                                purge_duration: x.purge_duration,
+                            };
+
+                            orders.perform_cmd(
+                                update_stratagem_button::update_stratagem(x)
+                                    .map_ok(|x| Msg::CmdSent(Box::new(x)))
+                                    .map_err(|x| Msg::CmdSent(Box::new(x))),
+                            );
+                        }
+                    }
+                    Command::Enable => {
+                        let m = config.create_enable_stratagem_model(model.fs.id);
+                        if let Some(x) = m {
+                            config.target_config = TargetConfig {
+                                interval: x.interval,
+                                report_duration: x.report_duration,
+                                purge_duration: x.purge_duration,
+                            };
+
+                            orders.perform_cmd(
+                                enable_stratagem_button::enable_stratagem(x)
+                                    .map_ok(|x| Msg::CmdSent(Box::new(x)))
+                                    .map_err(|x| Msg::CmdSent(Box::new(x))),
+                            );
+                        }
+                    }
+                    Command::Delete => {
+                        if let Some(x) = config.id {
+                            orders.perform_cmd(
+                                delete_stratagem_button::delete_stratagem(x)
+                                    .map_ok(|x| Msg::CmdSent(Box::new(x)))
+                                    .map_err(|x| Msg::CmdSent(Box::new(x))),
+                            );
+                        }
+                    }
+                }
+            }
+            Msg::CmdSent(fetch_object) => match fetch_object.response() {
+                Ok(response) => {
+                    log!("Response data: {:#?}", response.data);
+                    // Give feedback that stratagem has been set
+                    orders.skip();
+                }
+                Err(fail_reason) => {
+                    config.disabled = false;
+                    log!("Fetch error: {:#?}", fail_reason);
```

Comment:
```suggestion
                    error!("Fetch error: {:#?}", fail_reason);
```

And by the way, these macros do not do formatting, it will print `"Fetch error: {:#?}", fail_reason`

Reply:
Sure, good call.